compiler/rustc_borrowck/src/universal_regions.rs

```diff
@@ -674,7 +674,7 @@ impl<'cx, 'tcx> UniversalRegionsBuilder<'cx, 'tcx> {
                 // For a constant body, there are no inputs, and one
                 // "output" (the type of the constant).
                 assert_eq!(self.mir_def.did.to_def_id(), def_id);
-                let ty = tcx.type_of(self.mir_def.def_id_for_type_of());
+                let ty = tcx.bound_type_of(self.mir_def.def_id_for_type_of()).subst_identity();
```

Comment:
use `substs` from `DefiningTy::Const`? also `DefiningTy::FnDef` a few lines above also has `substs` that could be used on the `fn_sig`